Bottom: f0fcce7f458dec33f1190d5dd72075c4d8e0a34a
Top:    8b0b927a8b2dbea7e233300c165e817f47e7a521
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-24 17:14:03 +0800

Refresh of vhost-device-iotlb-api

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index fdae0e8..fcd81d7 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -36,6 +36,10 @@ static ushort max_mem_regions = 64;
 module_param(max_mem_regions, ushort, 0444);
 MODULE_PARM_DESC(max_mem_regions,
 	"Maximum number of memory regions in memory map. (default: 64)");
+static int max_iotlb_entries = 2048;
+module_param(max_iotlb_entries, int, 0444);
+MODULE_PARM_DESC(max_iotlb_entries,
+	"Maximum number of iotlb entries. (default: 2048)");
 
 enum {
 	VHOST_MEMORY_F_LOG = 0x1,
@@ -832,7 +836,7 @@ static int vhost_new_umem_range(struct vhost_umem *umem,
 	if (!node)
 		return -ENOMEM;
 
-	if (umem->numem == VHOST_IOTLB_SIZE) {
+	if (umem->numem == max_iotlb_entries) {
 		tmp = list_last_entry(&umem->umem_list, typeof(*tmp), link);
 		vhost_umem_free(umem, tmp);
 	}
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index 2154c25..807b532 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -141,8 +141,6 @@ struct vhost_virtqueue {
 	u32 busyloop_timeout;
 };
 
-#define VHOST_IOTLB_SIZE 2048
-
 struct vhost_msg_node {
   struct vhost_msg msg;
   struct vhost_virtqueue *vq;
